src/Integration/Service/SonarQubeServiceWrapper.cs

```diff
@@ -447,6 +462,58 @@ public Uri CreateProjectDashboardUrl(ConnectionInformation connectionInformation
 
         #endregion
 
+        #region Notifications
+
+        private static async Task<NotificationEvent[]> GetNotificationEvents(HttpClient client,
+            CancellationToken token, string projectKey, DateTimeOffset eventsSince)
+        {
+            var queryParameters = new Dictionary<string, object>
+            {
+                { "projects", projectKey },
+                { "from", ToJavaTimeFormat(eventsSince) }
+            };
+
+            var url = BuildNotificationsUrl(queryParameters);
+            var httpResponse = await InvokeGetRequest(client, url, token);
+            var stringResponse = await ReadResponse(httpResponse, token);
+
+            switch (httpResponse.StatusCode)
+            {
+                case HttpStatusCode.NotFound:
+                    return null;
+
+                case HttpStatusCode.OK:
+                    return ProcessJsonResponse<NotificationEvents>(stringResponse, token).Events
+                        ?? new NotificationEvent[0];
+
+                default:
+                    return new NotificationEvent[0];
+            }
+        }
+
+        private static string ToJavaTimeFormat(DateTimeOffset date)
+        {
+            // This is the only format the API accepts. ISO 8601 formats don't work.
+            // For example 2013-05-01T13:00:00+0100
+            var dateTime = date.ToString("yyyy-MM-ddTHH:mm:ss", CultureInfo.InvariantCulture);
+            var timezone = date.ToString("zzz", CultureInfo.InvariantCulture).Replace(":", "");
+
+            return dateTime + timezone;
+        }
+
+        private static string BuildNotificationsUrl(Dictionary<string, object> queryParameters)
```

Comment:
interesting.

Reply:
The good thing is that handles escaping.